### PR TITLE
Consider neutral checks as meeting Maestro policy

### PR DESCRIPTION
**Pull request by premun**
https://github.com/dotnet/arcade-services/issues/3636

### CODE REVIEW

**Comment by mmitche** [comment]
I think the way to fix this is just to remove AllChecksSuccessful altogether (or make it do nothing) and remove all the auto-merge capability.

**Comment by premun** [comment]
> I think the way to fix this is just to remove AllChecksSuccessful altogether (or make it do nothing) and remove all the auto-merge capability.

I will log an issue for this

**Comment by MiYanni** [comment]
@mmitche 

> I don't think that Neutral checks can be considered success. I think that this happens when the run hasn't started yet, but is expected based on GitHub required checks. If this is the case, then Maestro might try and merge.

Is `CheckConclusion` only after the check has been `Completed`? From the other PR conversation:

```
"checkRuns": {
  "nodes": [
    {
      "name": "sdk-unified-build",
      "status": "COMPLETED",
      "conclusion": "NEUTRAL"
    },
}
```

See how the `status` is `COMPLETED`? If the run hasn't started yet, the status wouldn't be `COMPLETED`, right? I don't know when `CheckConclusion`s are created.

TLDR: If `CheckConclusion` only is a result of `status` being `COMPLETED`, then this change *should* work, I believe.
